pkg/webhook/parser.go

```diff
@@ -440,6 +453,98 @@ func (Generator) RegisterMarkers(into *markers.Registry) error {
 	return nil
 }
 
+// FeatureGateMap represents enabled feature gates as a map for efficient lookup
+type FeatureGateMap map[string]bool
```

Comment:
The feature gate parsing logic appears to be needed in several places, should we create a centralised utility for this so that we only have to implement it once?

Reply:
yes totally agree, and adressed this